src/main/java/org/elasticsearch/index/translog/TranslogSnapshot.java

```diff
@@ -42,21 +42,19 @@
      * Create a snapshot of translog file channel. The length parameter should be consistent with totalOperations and point
      * at the end of the last operation in this snapshot.
      */
-    public TranslogSnapshot(List<ChannelSnapshot> orderedTranslogs, ESLogger logger) {
+    public TranslogSnapshot(List<Translog.Snapshot> orderedTranslogs) {
         this.orderedTranslogs = orderedTranslogs;
-        this.logger = logger;
         int ops = 0;
-        for (ChannelSnapshot translog : orderedTranslogs) {
+        for (Translog.Snapshot translog : orderedTranslogs) {
 
             final int tops = translog.estimatedTotalOperations();
             if (tops < 0) {
```

Comment:
Hmm can we change "tops < 0" to "tops == TranslogReader.UNKNOWN_OP_COUNT"?  And then assert tops > 0 otherwise?